products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -277,6 +277,45 @@ class CAPIBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
     }
   }
 
+  virtual void commandCannotBuildOutputDueToMissingInputs(Command* command,
+               BuildNode* outputNode, SmallPtrSet<Node*, 1> inputNodes) override {
+    if (cAPIDelegate.command_cannot_build_output_due_to_missing_inputs) {
+      llb_build_key_t output = { llb_build_key_kind_node,
+        strdup(outputNode->getName().str().c_str()) };
```

Comment:
We shouldn't use `strdup` here, this forces the client to free this memory.

I see now that we are doing this for `cycleDetected`, but I would rather we stop than add more instances. However, since that might be a more involved change can you just file a follow on bug for this?

Reply:
Will do.

Reply:
Actually, we can just free the memory after the delegate callback?

Reply:
Yup, that's right